internal/dag/builder.go

```diff
@@ -99,22 +100,22 @@ func (b *Builder) lookupService(m k8s.FullName, port intstr.IntOrString) *Servic
 
 	s := lookup()
 	if s != nil {
-		return s
+		return s, nil
 	}
 	svc, ok := b.Source.services[m]
 	if !ok {
-		return nil
+		return nil, fmt.Errorf("service %q not found", m.Name)
```

Comment:
```suggestion
		return nil, fmt.Errorf("service %q not found", m)
```

Let's print the fully qualified name to remove any possible ambiguity.